lib/agent0/agent0/hyperdrive/policies/random_hold.py

```diff
@@ -0,0 +1,313 @@
+        """
+        # We scan open positions and select a long that's ready to be closed
+        longs_ready_to_close: list[RandomHold._Position] = [
+            position
+            for position in self.open_positions
+            if position.ready_to_close and position.action_type == HyperdriveActionType.OPEN_LONG
+        ]
+        # Sanity check, we should have at least one, otherwise close long wouldn't be an available action
+        assert len(longs_ready_to_close) > 0
+
+        # Select a random one
+        long_to_close = longs_ready_to_close[self.rng.integers(len(longs_ready_to_close))]
+        # Set flag that the transaction was sent
+        long_to_close.txn_sent = True
+
+        ignore_slippage = self.rng.choice([True, False], size=1) if self.randomly_ignore_slippage_tolerance else False
+        if ignore_slippage:
+            slippage = None
+        else:
+            slippage = self.slippage_tolerance
+        return [close_long_trade(long_to_close.bond_amount, long_to_close.maturity_time, slippage)]
+
+    def close_random_short(
+        self, interface: HyperdriveReadInterface, wallet: HyperdriveWallet
+    ) -> list[Trade[HyperdriveMarketAction]]:
+        """Closes a random short that's ready to be closed.
+
+        Arguments
+        ---------
+        interface: HyperdriveReadInterface
+            Interface for the market on which this agent will be executing trades (MarketActions).
+        wallet: HyperdriveWallet
+            The agent's wallet.
+
+        Returns
+        -------
+        list[Trade[HyperdriveMarketAction]]
+            A list with a single Trade element for closing a Hyperdrive short.
+        """
+        # We scan open positions and select a short that's ready to be closed
+        shorts_ready_to_close: list[RandomHold._Position] = [
+            position
+            for position in self.open_positions
+            if position.ready_to_close and position.action_type == HyperdriveActionType.OPEN_SHORT
+        ]
+        # Sanity check, we should have at least one, otherwise close short wouldn't be an available action
+        assert len(shorts_ready_to_close) > 0
+
+        # Select a random one
+        short_to_close = shorts_ready_to_close[self.rng.integers(len(shorts_ready_to_close))]
+        # Set flag that the transaction was sent
+        short_to_close.txn_sent = True
+
+        ignore_slippage = self.rng.choice([True, False], size=1) if self.randomly_ignore_slippage_tolerance else False
+        if ignore_slippage:
+            slippage = None
+        else:
+            slippage = self.slippage_tolerance
+        return [close_short_trade(short_to_close.bond_amount, short_to_close.maturity_time, slippage)]
+
+    def post_action(self, interface: HyperdriveReadInterface, trade_results: list[TradeResult]) -> None:
+        """Function that gets called after actions have been executed. This allows the policy
```

Comment:
ok but what is this fn actually doing then? :P

Reply:
The `post_action` function was added in a separate PR here: https://github.com/delvtech/agent0/pull/1320
The comments here do say what this function is doing in `post_action`, but I'll be sure to highlight it here